pkg/tempo/tempo.go

```diff
@@ -0,0 +1,205 @@
+	tempo.metricViews, err = newMetricViews()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create metric views %w", err)
+	}
+
+	createCtx := context.Background()
+	err = tempo.buildAndStartPipeline(createCtx, cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create exporter %w", err)
+	}
+
+	return tempo, nil
+}
+
+// Stop stops the OpenTelemetry collector subsystem
+func (t *Tempo) Stop() {
+	shutdownCtx := context.Background()
+
+	if err := t.receivers.ShutdownAll(shutdownCtx); err != nil {
+		t.logger.Error("failed to shutdown receiver", zap.Error(err))
+	}
+
+	if err := t.pipelines.ShutdownProcessors(shutdownCtx); err != nil {
+		t.logger.Error("failed to shutdown processors", zap.Error(err))
+	}
+
+	if err := t.receivers.ShutdownAll(shutdownCtx); err != nil {
+		t.logger.Error("failed to shutdown receivers", zap.Error(err))
+	}
+
+	view.Unregister(t.metricViews...)
+}
+
+func (t *Tempo) buildAndStartPipeline(ctx context.Context, cfg Config) error {
+	// create component factories
+	otelConfig, err := cfg.otelConfig()
+	if err != nil {
+		return fmt.Errorf("failed to load otelConfig from agent tempo config %w", err)
+	}
+
+	factories, err := tracingFactories()
+	if err != nil {
+		return fmt.Errorf("failed to load tracing factories %w", err)
+	}
+
+	// start exporter
+	t.exporter, err = builder.NewExportersBuilder(t.logger, otelConfig, factories.Exporters).Build()
+	if err != nil {
+		return fmt.Errorf("failed to build exporters %w", err)
+	}
+
+	err = t.exporter.StartAll(ctx, t)
+	if err != nil {
+		return fmt.Errorf("failed to start exporters %w", err)
+	}
+
+	// start pipelines
+	t.pipelines, err = builder.NewPipelinesBuilder(t.logger, otelConfig, t.exporter, factories.Processors).Build()
+	if err != nil {
+		return fmt.Errorf("failed to build exporters %w", err)
+	}
+
+	err = t.pipelines.StartProcessors(ctx, t)
+	if err != nil {
+		return fmt.Errorf("failed to start processors %w", err)
+	}
+
+	// start receivers
+	t.receivers, err = builder.NewReceiversBuilder(t.logger, otelConfig, t.pipelines, factories.Receivers).Build()
+	if err != nil {
+		return fmt.Errorf("failed to start receivers %w", err)
+	}
+
+	err = t.receivers.StartAll(ctx, t)
+	if err != nil {
+		return fmt.Errorf("failed to start receivers %w", err)
+	}
+
+	return nil
+}
+
+// ReportFatalError implements component.Host
+func (t *Tempo) ReportFatalError(err error) {
+	t.logger.Error("fatal error reported", zap.Error(err))
+}
+
+// GetFactory implements component.Host
+func (t *Tempo) GetFactory(kind component.Kind, componentType configmodels.Type) component.Factory {
+	return nil
+}
+
+// GetExtensions implements component.Host
+func (t *Tempo) GetExtensions() map[configmodels.Extension]component.ServiceExtension {
+	return nil
+}
+
+// GetExporters implements component.Host
+func (t *Tempo) GetExporters() map[configmodels.DataType]map[configmodels.Exporter]component.Exporter {
+	return nil
+}
+
+func newLogger(level logging.Level) *zap.Logger {
```

Comment:
Can we do something so there's always a `component=tempo` KVP in the logs from here?